Userland/Libraries/LibGL/SoftwareGLContext.cpp

```diff
@@ -76,12 +96,24 @@ void SoftwareGLContext::gl_clear_color(GLclampf red, GLclampf green, GLclampf bl
 
 void SoftwareGLContext::gl_color(GLdouble r, GLdouble g, GLdouble b, GLdouble a)
 {
+    if (should_append_to_listing()) {
+        append_to_listing<&SoftwareGLContext::gl_color>(r, g, b, a);
+        if (!should_execute_after_appending_to_listing())
+            return;
+    }
```

Comment:
Can we somehow factor out all these append list blocks? Maybe with a macro?

Reply:
If we're okay with the template magic, I think the better option would be to add a layer between `GLContext` and `SoftwareGLContext` that handles this, and move all this list logic there
That would simplify specific context code, _and_ will be basically plug&play for future context impls

Reply:
Except the one case that needs to allocate and copy something, I converted everything to use a macro `APPEND_TO_CALL_LIST_AND_RETURN_IF_NEEDED(fn, args...)`.